### PR TITLE
feat: update v11 step 1 to React 18

### PR DESCRIPTION
**Pull request by alisonjoseph**
Ref #https://github.com/carbon-design-system/carbon/issues/13630

#### Changelog

**New**

- Add react testing library

**Changed**

- Upgrade to React 18
- Update index.js to use `createRoot`

**Removed**

- remove enzyme
- remove enzyme-adapter-react-16

**Note: Tests should fail, they will only pass after the user finishes step 1 and adds the header.**

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
   });
 
-  const wrapper = shallow(<App />);
-  it('contains a TutorialHeader', () => {
-    expect(wrapper.find('TutorialHeader').length).toBe(1);
+  it('contains Carbon Tutorial heading', () => {
+    render(<App />);
+    expect(screen.getByText('Carbon Tutorial')).toBeInTheDocument();
   });
 });
```
Comment (francinelucca) [inline src/App.test.js:15]: I'm getting an error when trying to run this on the step 2 branch 🤔
image

**Review by francinelucca — CHANGES_REQUESTED** [review]
Do we need to update `src/index.js` to use `createRoot`? same comment for all steps 
image

https://stackoverflow.com/questions/71668256/deprecation-notice-reactdom-render-is-no-longer-supported-in-react-18

File: src/App.test.js (hunk 1 above)
Comment (alisonjoseph) [reply src/App.test.js:15]: Should be fixed now 🙏

**Review by francinelucca — APPROVED** [review]
🚀

**Pull request merged**